### ISSUE TITLE
Adding plot_title to normal plots

### ISSUE THREAD

**Issue by TheFlike** [N2-Survey/SurveyFramework#105]
## Description

Add `plot_title` variable to simple plot functions in n2survey/plot/plot_multiple_choice.py, n2survey/plot/plot_likert.py and n2survey/plot/bar.py

- [ ] use variable `plot_title=True` for adding the question as plot-title,
- [ ]  `plot_title=False` should restrict displaying of any title over the plot 
- [ ] For ` plot_title='my own title'` it should write "my own title" as title.

**Comment by NiRuff** [comment]
I should be done with that one now - however, I saw that each of these three functions have some distinct title functionalities - Is it okay to delete the old ones to only have one default behavior now or should the old options be kept?
In the current version I commented them out and did some testing - I did not run into any problems as I tried to exchange the old usages of these parameters by the new more standardized appraoch using the plot_title parameter.

**Comment by NiRuff** [comment]
Yes exactly! I removed it now and pushed everythin - however, I get some error in the testing that is automatically performed in the pull request but can not really derive what is causing the error.

**Comment by TheFlike** [comment]
The error says you didn't run pre-commit, before git commit the environment should run
autoflake
isort
black
flake8

normally just the command "pre-commit" runs these.

**Comment by NiRuff** [comment]
Hey,

sorry I am a bit lost on that pre-commit - where does it come from?
I usually push my projects using the git plugin in PyCharm and also did that here for this project.
Is pre-commit related to git?
Or is it related to some python package?

And what is the issue with it? It is just checking if e.g. line lengths are according to some formatting rules?

Best,
Nicolas

**Comment by TheFlike** [comment]
I guess with your environment you mean the n2survey environment? Else you'd have to install it there as well.
I don't know anything about PyCharm and I normally just use terminal, so I can't help you with PyCharm specifically.

But if poetry is installed in the n2survey environment then just using `git add -A `and the `pre-commit` command in terminal (the same folder and environment you already used to install pre-commit) should work and reformat your code, if necessary.

**Comment by NiRuff** [comment]
Yes exactely, the n2survey environment.

I tried to push via the git command in the command prompt instead of PyCharm now and had an error when running pre-commit.
adding using git add -A worked, then I got the following error:

![grafik](https://user-images.githubusercontent.com/50486014/172384909-4f5f07fd-a5d6-4e12-bb49-ec2dc1c20ab7.png)


I only posted the beginning of the error message here, the whole error was very long

Unfortunately this error message is not very helpful in fixing whatever the issue is - I'll also put some more effort in trying to fix it, but it is somewhat dissappointing spending way more time trying to fix these autoflake nad pre-commit problems than writing the code itself :/

But thanks for the help so far, though!

**Comment by TheFlike** [comment]
Yes, I've heard that more then once, another colleague also had a problem with the environment, he recreated it with the CONTRIBUTING.md guideline and then everything worked. But Yeah its quite annoying in the beginning.

There seems to be always a little bit of a hassle when first time using the pre-commit hooks... on the other hand, at least in my experience, when it worked once it worked always ever since and I liked the formatting and the restrictions on making functions not too complicated so much that I added it to my own programming projects as well....
At least I can tell that it most probably will work in the end, from my sample size of now (including you and me) 4 people who had problems with this :D.

**Comment by NiRuff** [comment]
Hey,

sorry for the late reply - I was participating in a conference last week.

I could solve this pre-commit issue now.
However, I run into an error now, that is a bit of a problem... the testing script is using some of the old unharmonized properties for adding plot title.
This issue was about making the syntax analogous for all the plots, therefor I used the heteogeneous parameters for adding titles in the past.
However, the testing script is using e.g. the `display_title `parameter, that I substituted by the wished parameter `plot_title` (see initial description of the issue).
Therefor, by definition of this issue here, I will run into errors in the tests, that are adapted to the old versions for plotting titles.

Best,
Nicolas

**Comment by NiRuff** [comment]
I changed the parameter in the test now and everything runs through. Should I merge it myself or wait for you to do it?

Best,
Nicolas
